Adding TS files as types is non-standard and prevent uses of it as TS modules

Hello back, 
I prepared a PR to fix this issue, I will open it if you agree with this changes.

### Current behaviour

Using this module with typescript and `"moduleResolution": "node"` throws two errors:

**The first.**
```
node_modules/bitly/src/lib.ts:2:21 - error TS7016: Could not find a declaration file for module 'request-promise'. '/.../project/node_modules/request-promise/lib/rp.js' implicitly has an 'any' type.
  Try `npm install @types/request-promise` if it exists or add a new declaration (.d.ts) file containing `declare module 'request-promise';`

2 import request from 'request-promise';
                      ~~~~~~~~~~~~~~~~~
```

This explains because `tsc` with the node module resolution checks types of imported modules.
However, this module exports the TS code and not the rendered declaration, hence it crawls the whole files and types (which are added as dev deps).

So, I think there are two solutions:

1. Moving the `@types/request-promise` to deps;
2. Generating the declaration files (`*.d.ts`) instead of using the TS code.

**The second.**
```
node_modules/bitly/src/lib.ts:96:28 - error TS2532: Object is possibly 'undefined'.

96     item => (isUri(item) ? result.shortUrl.push(item) : typeof item === 'string' && result.hash.push(item))
                              ~~~~~~~~~~~~~~~

node_modules/bitly/src/lib.ts:96:85 - error TS2532: Object is possibly 'undefined'.

96     item => (isUri(item) ? result.shortUrl.push(item) : typeof item === 'string' && result.hash.push(item))
                                                                                       ~~~~~~~~~~~
```

I don't know why I still have these two errors, it should have been fixed by adding these lines in #63 :
https://github.com/tanepiper/node-bitly/blob/acce1a6b87a9ba2d69f9e60b10c8965fcb37f0ec/src/lib.ts#L93-L94
I think it's due to my tsc configs.

These two error makes me think that the second option is better as such cases cannot happen from pure `*.d.ts`.

### Desired behaviour

Doesn't throw errors of compilation when used as a dependency.

### References

In the [TS Handbook](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html), the advised way to publish is to use the `--declaration` flag, which will render the `*.d.ts` files.

Also, in an [old article](https://medium.com/@mweststrate/how-to-create-strongly-typed-npm-modules-1e1bda23a7f4):

> It is recommended to specify ‘files’ in your package.json (or use a .npmignore file) to keep your TypeScript sources outside the actual package that will be submitted to npm. As rule of thumb: *.ts files should be in version control but not in the package, *.d.ts and *.js files should be in the package but not in version control.

--- a/.npmignore
+++ b/.npmignore
@@ -1,4 +1,14 @@
+# CI
 .circleci/
+
+# Editor
 .vscode
+
+# Web Pages & docs
 fixtures/
 _config.yml
+
+# TS Files
+tsconfig.json
+test
+src/*

--- a/global.d.ts
+++ b/global.d.ts
@@ -1,3 +1,0 @@
-export * from './src/bitly.d';
-export * from './src/bitly';
-export * from './src/lib';

--- a/package.json
+++ b/package.json
@@ -69,7 +69,7 @@
     "typedoc": "^0.11.1",
     "typescript": "^2.9.1"
   },
-  "typings": "./global.d.ts",
+  "typings": "./dist/*.d.ts",
   "dependencies": {
     "request": "^2.87.0",
     "request-promise": "^4.2.2",

--- a/src/bitly.spec.ts
+++ b/src/bitly.spec.ts
@@ -8,7 +8,7 @@ const EXAMPLE_URL_BITLY = 'http://bit.ly/2hpSRbP';
 import '../test/bootstrap';
 
 import { BitlyClient } from './bitly';
-import { BitlyError } from './bitly.d';
+import { BitlyError } from './bitly.types';
 
 describe('Bitly client', () => {
   let bitly: BitlyClient;

--- a/src/bitly.ts
+++ b/src/bitly.ts
@@ -1,5 +1,5 @@
 import { doRequest, sortUrlsAndHash } from './lib';
-import { BitlyConfig, BitlyResponse, BitlyError, BitlyUrlQueryParams, BitlyResponseData } from './bitly.d';
+import { BitlyConfig, BitlyResponse, BitlyError, BitlyUrlQueryParams, BitlyResponseData } from './bitly.types';
 
 /**
  *

--- a/src/bitly.types.ts
+++ b/src/bitly.types.ts
@@ -26,7 +26,7 @@ export interface ExpandInteface {
   expand: string[];
 }
 
-type BitlyResponseData = ShortenResponse | ExpandInteface;
+export type BitlyResponseData = ShortenResponse | ExpandInteface;
 
 export interface BitlyResponse {
   status_code: number;

--- a/src/lib.ts
+++ b/src/lib.ts
@@ -3,7 +3,7 @@ import request from 'request-promise';
 
 const isUri = require('valid-url').isUri;
 
-import { BitlyConfig, BitlyUrlQueryParams, BitlyResponse } from './bitly.d';
+import { BitlyConfig, BitlyUrlQueryParams, BitlyResponse } from './bitly.types';
 
 /**
  * The internal library of node-bitly

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "module": "commonjs",
+    "declaration": true,
     "esModuleInterop": true,
     "target": "es6",
     "noImplicitAny": true,

